src/ListContainer.hpp

```diff
@@ -39,6 +39,65 @@ struct ipl_rangestruct {
 }
 #endif
 
+// class for linking IPs to filter groups, complete with comparison operators
+// allowing standard C++ sort to work
+class ipmap
+{
+public:
+    ipmap(uint32_t a, String g)
+    {
+        addr = a;
+        group = g;
+    };
+    uint32_t addr;
+    String group;
+    int operator<(const ipmap &a) const
+    {
+        return addr < a.addr;
+    };
+    int operator<(const uint32_t &a) const
+    {
+        return addr < a;
+    };
+    int operator==(const uint32_t &a) const
+    {
+        return a == addr;
+    };
+};
+
+class datamap
+{
+public:
+    datamap(String &k, String &g){
+        key = k;
+        group = g;
+    };
+    String key;
+    String group;
+    int operator<(const datamap &a) const
+    {
+        return key.compare(a.key);
+    };
+    int operator==(const String &a) const
+    {
+        if( key.compare(a) == 0) return 1;
+        return 0;
+    };
+};
+
+// structs linking subnets and IP ranges to filter groups
+struct subnetstruct {
+    uint32_t maskedaddr;
+    uint32_t mask;
+    String group;
+};
+
+struct rangestruct {
+    uint32_t startaddr;
+    uint32_t endaddr;
```

Comment:
Codacy found an issue: [struct member 'rangestruct::endaddr' is never used.](https://app.codacy.com/manual/numsys/e2guardian/pullRequest?prid=6238856)